code/game/turfs/turf_texture.dm

```diff
@@ -21,18 +21,20 @@
 //Effect object we use to hold our groceries
 /obj/effect/turf_texture
 	mouse_opacity = MOUSE_OPACITY_TRANSPARENT
+	blend_mode = BLEND_MULTIPLY
+	plane = FLOOR_PLANE
+	appearance_flags = KEEP_TOGETHER
 	///Associated texture
 	var/datum/turf_texture/parent_texture
 
 /obj/effect/turf_texture/Initialize(mapload, datum/turf_texture/_texture)
```

Comment:
I'd much rather you make `turf_texture` a system that just adds overlays to the turf. Having it as an `/obj/effect` adds a lot of overhead.

Reply:
That or repath it to `/atom/movable`, since it is not physical (/`obj` is pretty full of random BS)

Reply:
I'd like to opt for ``/atom/movable`` for plans I have later. If it's a single 'thing' copy-pasted everywhere in vis contents, I have a little more control in terms of actively editing it, for a 'dirt' system later.

Reply:
Oh, is it one object per texture? i wonder if it's faster to do that + vis_contents, although it won't work with zmimic

Reply:
zmimic zchmimick